OTAnalytics/plugin_ui/nicegui_gui/ui_factory.py

```diff
@@ -88,24 +90,50 @@ async def askopenfilenames(
         title: str,
         filetypes: Iterable[tuple[str, str | list[str] | tuple[str, ...]]],
         extension_options: dict[str, list[str] | None] | None = None,
-    ) -> Literal[""] | tuple[str, ...]:
+    ) -> list[Path]:
         # For now, we'll just support selecting a single file
         # In a real implementation, this would allow selecting multiple files
         # Convert list/tuple of extensions to individual (desc, ext) pairs
         converted_filetypes = []
         for desc, ext in filetypes:
             if isinstance(ext, str):
-                converted_filetypes.append((desc, ext))
+                converted_filetypes.append(ext)
             elif isinstance(ext, (list, tuple)) and ext:
                 # Use the first extension from the list/tuple
-                converted_filetypes.append((desc, ext[0]))
+                converted_filetypes.append(ext[0])
 
-        file_path = await self.askopenfilename(
-            title, converted_filetypes, "", extension_options
+        file_paths = await LocalFilePicker(
+            directory=Path.home(),
+            show_hidden_files=False,
+            show_files_only_of_type=None,  # Show all files
+            show_only_directories=False,
+            extension_options=extension_options,
+            multiple=True,
         )
-        if file_path:
-            return (file_path,)
-        return ""
+        return self.get_all_files_with_correct_file_ending_in_directory(
+            file_paths, converted_filetypes
+        )
+
+    def get_all_files_with_correct_file_ending_in_directory(
```

Comment:
Extract this method into a helper function in `OTAnalytics/helpers/file_helpers.py`. Reuse it also at its original copy place.